src/playlist.js

```diff
@@ -4,16 +4,48 @@ const url = require('url');
 const api = require('./api');
 
 /**
+ * Sorts descending based on a given property
+ */
+function statisticsComparator(prop, a, b) {
+    let aProp = a.statistics[prop];
+    let bProp = b.statistics[prop];
+
+    if (isNaN(aProp)) {
+      aProp = 0;
+    }
+
+    if (isNaN(bProp)) {
+      bProp = 0;
+    }
+
+    return bProp - aProp;
+}
+
+/**
  * Sorts descending based on viewCount / popularity
  */
-function comparator(a, b) {
-  return b.statistics.viewCount - a.statistics.viewCount;
+function viewComparator(a, b) {
+  return statisticsComparator('viewCount', a, b);
+}
+
+function likeComparator(a, b) {
+  return statisticsComparator('likeCount', a, b);
 }
 
-async function getSortedPlaylist(playListId) {
-  const videoIds = await api.getVideoIdsFromPlayList(playListId);
-  const videos = await Promise.all(videoIds.map(api.getVideoDetails));
-  return videos.sort(comparator);
+/**
+ * Returns a sorted playlist based on a given comparator.
+ * @param playListId
+ * @param {function} comparator - the comparator to sort by.
+ * @returns {Promise.<Array.<T>|*>}
+ */
+async function getSortedPlaylist(playListId, comparator = viewComparator) {
+  try {
```

Comment:
I'm not sure why a `try/catch` is required over here?
Because there's a `try/catch` in the [main function](https://github.com/ArcticZeroo/youtube-playlist-sorter/blob/master/src/index.js#L41)

Reply:
Node's async doesn't catch errors thrown by `await`, they need to be explicitly thrown with `throw` or explicitly caught in some way. As a result, if the YouTube API replies with an error of some kind, an error will be printed to console about an uncaught promise, but getSortedPlaylist won't actually reject the promise.

Reply:
Are you sure about that?? 🤔
Because I commented out the `try/catch` added by you [over here](https://github.com/ArcticZeroo/youtube-playlist-sorter/blob/master/src/playlist.js#L42) and explicitly threw an error in [api.getVideoIdsFromPlayList](https://github.com/ArcticZeroo/youtube-playlist-sorter/blob/master/src/api.js#L15) and it did get caught by the catch block in [main.js](https://github.com/ArcticZeroo/youtube-playlist-sorter/blob/master/src/index.js#L52)

Reply:
I'm like 75% sure (it's definitely been an issue many times for me on other projects), I'll do some testing of my own when I'm back at my dorm